src/functions/utils/rxjs-dynamic-duration.ts

```diff
@@ -44,12 +44,55 @@ export const rxjsDynamicDuration = <T>(
 	mapDuration: (lastDuration: number, event: T, index: number) => number,
 	initialDuration = 0,
 ) => {
-	const mappableDuration = createMappableValue(mapDuration, initialDuration);
```

Comment:
I removed the mappable value stuff because tracking the index and previous value is simpler.